python/lsst/ip/isr/linearize.py

```diff
@@ -679,6 +680,52 @@ def __call__(self, image, **kwargs):
             return False, 0
 
 
+class LinearizeSpline(LinearizeBase):
+    """Correct non-linearity with a spline model.
+
+    corrImage = uncorrImage - Spline(coeffs, uncorrImage)
+
+    Notes
+    -----
+
+    The spline fit calculates a correction as a function of the
+    expected linear flux term.  Because of this, the correction needs
+    to be subtracted from the observed flux.
+
+    """
+    LinearityType = "Spline"
+
+    def __call__(self, image, **kwargs):
+        """Correct for non-linearity.
+
+        Parameters
+        ----------
+        image : `lsst.afw.image.Image`
+            Image to be corrected
+        kwargs : `dict`
+            Dictionary of parameter keywords:
+            ``"coeffs"``
+                Coefficient vector (`list` or `numpy.array`).
+            ``"log"``
+                Logger to handle messages (`lsst.log.Log`).
+
+        Returns
+        -------
+        output : `bool`
+            If true, a correction was applied successfully.
+        """
+        splineCoeff = kwargs['coeffs']
+        centers, values = np.split(splineCoeff, 2)
+        interp = afwMath.makeInterpolate(centers.tolist(), values.tolist(),
+                                         afwMath.stringToInterpStyle("AKIMA_SPLINE"))
+
+        ampArr = image.getArray()
+        delta = interp.interpolate(ampArr.flatten())
+        ampArr -= np.array(delta).reshape(ampArr.shape)
+
+        return True, 0
```

Comment:
The fact that you return something saying whether it was applied successfully implies that you might return something else, so I expected to see, from the docs/form of API, to maybe see a copy, a try, and a finally resetting to the copy on fail, and a return `False`. That seems expensive and probably unnecessary, but just saying what I'd expect to happen given that it says you get a return value if successful here.

Reply:
The numeric linearizers (spline/squared/polynomial) have no way to fail.  The lookup table can fail, if all pixel values fall outside the lookup table range.  This might be an issue with splines, although the low-end of the fit is padded to ensure a linear extrapolation in that direction.